File: Examples/DotNet/CSharp/Pictures/AddPicturesWorksheet.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace Pictures
{
    class AddPicturesWorksheet
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample_Test_Book.xls";
            string sheetName = "Sheet6";
            int? upperLeftRow = 5;
            int? upperLeftColumn = 5;
            int? lowerRightRow = 10;
            int? lowerRightColumn = 10;
            string picturePath = "aspose-cloud.png";
            string storage = null;
            string folder = null;
            byte[] file = null;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, null, null, System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                storageApi.PutCreate(picturePath, null, null, System.IO.File.ReadAllBytes(Common.GetDataDir() + picturePath));

                // Invoke Aspose.Cells Cloud SDK API to add picture to worksheet
                PicturesResponse apiResponse = cellsApi.PutWorksheetAddPicture(fileName, sheetName, upperLeftRow, upperLeftColumn, lowerRightRow, lowerRightColumn, picturePath, storage, folder, file);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Add Pictures to Excel Worksheet, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Node.js/SDK/Pictures/AddPicturesWorksheet.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var name = "Sample_Test_Book.xls";
var sheetName = "Sheet6";
var upperLeftRow = 5;
var upperLeftColumn = 5;
var lowerRightRow = 10;
var lowerRightColumn = 10;
var picturePath = "aspose-cloud.png";
try {
// Upload source file to aspose cloud storage
	storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');

		storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');

			// Invoke Aspose.Cells Cloud SDK API to add a picture to a worksheet
			cellsApi.PutWorksheetAddPicture(name, sheetName, upperLeftRow, upperLeftColumn, lowerRightRow, lowerRightColumn, picturePath, null, null, null, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');
				
				// Download updated Workbook from storage server
				storageApi.GetDownload(name, null, null, function(responseMessage) {
					assert.equal(responseMessage.status, 'OK');
					var outfilename = name;
					var writeStream = fs.createWriteStream(data_path + "_out_" + outfilename);
					writeStream.write(responseMessage.body);
					});
				});
			});
		});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1